pkg/controllers/customroute/custom_route_controller.go

```
@@ -0,0 +1,307 @@
+		destSecret: types.NamespacedName{
+			Namespace: destSecretNamespace,
+			Name:      destSecretName,
+		},
+		componentRoute: types.NamespacedName{
+			Namespace: componentRouteNamespace,
+			Name:      componentRouteName,
+		},
+		ingressLister:  ingressInformer.Lister(),
+		ingressClient:  ingressClient,
+		routeLister:    routeInformer.Lister(),
+		routeClient:    routeSetter,
+		secretLister:   kubeInformersForNamespaces.SecretLister(),
+		operatorClient: operatorClient,
+		resourceSyncer: resourceSyncer,
+	}
+
+	return factory.New().
+		WithInformers(
+			ingressInformer.Informer(),
+			routeInformer.Informer(),
+			kubeInformersForNamespaces.InformersFor("openshift-config").Core().V1().Secrets().Informer(),
+			kubeInformersForNamespaces.InformersFor("openshift-authentication").Core().V1().Secrets().Informer(),
+		).
+		WithSyncDegradedOnError(operatorClient).
+		WithSync(controller.sync).
+		ResyncEvery(1*time.Minute).
+		ToController("customRouteController", eventRecorder)
+}
+
+func (c *customRouteController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	foundConditions := []operatorv1.OperatorCondition{}
+
+	ingressConfig, ingressConditions := common.GetIngressConfig(c.ingressLister, "IngressConfig")
+	foundConditions = append(foundConditions, ingressConditions...)
+
+	// Copy the ingressConfig object
+	ingressConfigCopy := ingressConfig.DeepCopy()
+
+	// Create or modify the router
+	if len(foundConditions) == 0 {
+		foundConditions = append(foundConditions, c.handleRoute(ctx, ingressConfigCopy)...)
+	}
+
+	// Copy source secret to destination
+	if len(foundConditions) == 0 {
+		foundConditions = append(foundConditions, c.syncSecret(ingressConfigCopy)...)
+	}
+
+	return common.UpdateControllerConditions(c.operatorClient, knownConditionNames, foundConditions)
+}
+
+func (c *customRouteController) handleRoute(ctx context.Context, ingress *configv1.Ingress) (conditions []operatorv1.OperatorCondition) {
```

I think we will eventually want @deads2k to bless this PR and I agree with him that the pattern when conditions are returned is making the code hard to read. Maybe just return an error?

---

s/handleRoute/applyRoute, pass the expected route as an argument